*** In the package MASS you’ll find the data set anorexia, which contains data on pre- and post-treatment weights (in pounds) of 72 young women suffering from the disease, obtained from Hand et al. (1994). One group of women is the control group (in other words, no intervention), and the other two groups are the cognitive behavioral program and family support intervention program groups. Load the library and ensure you can access the data frame and understand its contents. Let µd denote the mean difference in weight, computed as (post-weight − pre-weight).*** 

***  a. Regardless of which treatment group the participants fall into, conduct and conclude an appropriate hypothesis test with α = 0.05 for the entire set of weights for the following hypotheses: *** 

H0 : µd = 0

HA : µd > 0


In [10]:
library("MASS")
anorexia[1:2,]
n <- nrow(anorexia)

(sd(anorexia$Postwt))^2 - (sd(anorexia$Prewt))^2

Treat Prewt Postwt
1 Cont  80.7  80.2  
2 Cont  89.4  80.1

[1] "CBT"  "Cont" "FT"

[1] 37.70605

Since the different in variance is different than 0. So we use unpaired/independent sample

In [18]:
t.test(x=anorexia$Postwt, y=anorexia$Prewt, alternative = "greater", conf.level = .95)


	Welch Two Sample t-test

data:  anorexia$Postwt and anorexia$Prewt
t = 2.4528, df = 121.36, p-value = 0.007798
alternative hypothesis: true difference in means is greater than 0
95 percent confidence interval:
 0.8961579       Inf
sample estimates:
mean of x mean of y 
 85.17222  82.40833 


Since p-value is << 0.05, hence we can reject Ho.


***  b. Next, conduct three separate hypothesis tests using the same defined hypotheses, based on which treatment group the participants fall into. What do you notice? ***




In [31]:
cat("Unpaired/Independent Samples: Unpooled Variances" )
t.test(x=anorexia$Postwt, y=anorexia$Prewt, alternative = "greater", conf.level = .95)

cat("==========================================================================\n")
cat("Unpaired/Independent Samples: Pooled Variance")
t.test(x=anorexia$Postwt, y=anorexia$Prewt, alternative = "greater", conf.level = .95, var.equal  = TRUE)

cat("==========================================================================\n")
cat("Paired/Dependent Samples")
t.test(x=anorexia$Postwt,y=anorexia$Prewt,alternative="greater",conf.level=0.95, paired=TRUE)


Unpaired/Independent Samples: Unpooled Variances


	Welch Two Sample t-test

data:  anorexia$Postwt and anorexia$Prewt
t = 2.4528, df = 121.36, p-value = 0.007798
alternative hypothesis: true difference in means is greater than 0
95 percent confidence interval:
 0.8961579       Inf
sample estimates:
mean of x mean of y 
 85.17222  82.40833 


Unpaired/Independent Samples: Pooled Variance


	Two Sample t-test

data:  anorexia$Postwt and anorexia$Prewt
t = 2.4528, df = 142, p-value = 0.007693
alternative hypothesis: true difference in means is greater than 0
95 percent confidence interval:
 0.8982443       Inf
sample estimates:
mean of x mean of y 
 85.17222  82.40833 


Paired/Dependent Samples


	Paired t-test

data:  anorexia$Postwt and anorexia$Prewt
t = 2.9376, df = 71, p-value = 0.002229
alternative hypothesis: true difference in means is greater than 0
95 percent confidence interval:
 1.195825      Inf
sample estimates:
mean of the differences 
               2.763889 


All of the p-value are less than the significant level. Hence we can conclude that the true difference in means is greater than 0

*** Another ready-to-use data set in R is PlantGrowth (Dobson, 1983), which records a continuous measure of the yields of a certain plant, looking at the potential effect of two supplements administered during growth to increase the yield when compared to a control group with no supplement.*** 

***  c. Set up hypotheses to test whether the mean yield for the control group is less than the mean yield from a plant given either of the treatments. Determine whether this test should proceed using a pooled estimate of the variance or whether Welch’s t-test would be more appropriate. *** 

***  d. Conduct the test and make a conclusion (assuming normality of the raw observations).*** 


In [13]:
PlantGrowth[1:2,]
cat("Variance\n")
tapply(PlantGrowth$weight, INDEX = PlantGrowth$group, FUN = var)
cat("Mean\n")
tapply(PlantGrowth$weight, INDEX = PlantGrowth$group, FUN = mean)


cat("Since the variance in three groups are not equal. We should use Unpaired/Independent Samples: Unpooled Variances\n")
#Ho:  ctrl- trt =0 
#Ha: ctrl - trt < 0


cat("Crtl vs. Trt1 \n")
t.test(x=PlantGrowth$weight[PlantGrowth$group == "ctrl"]
       , y=PlantGrowth$weight[PlantGrowth$group == "trt1"], alternative = "less", conf.level = .95)

cat("\n\nCrtl vs. Trt2 \n")
t.test(x=PlantGrowth$weight[PlantGrowth$group == "ctrl"]
       , y=PlantGrowth$weight[PlantGrowth$group == "trt2"], alternative = "less", conf.level = .95)

weight group
1 4.17   ctrl 
2 5.58   ctrl

Variance


ctrl      trt1      trt2 
0.3399956 0.6299211 0.1958711

Mean


ctrl  trt1  trt2 
5.032 4.661 5.526

Since the variance in three groups are not equal. We should use Unpaired/Independent Samples: Unpooled Variances
Crtl vs. Trt1 



	Welch Two Sample t-test

data:  PlantGrowth$weight[PlantGrowth$group == "ctrl"] and PlantGrowth$weight[PlantGrowth$group == "trt1"]
t = 1.1913, df = 16.524, p-value = 0.8748
alternative hypothesis: true difference in means is less than 0
95 percent confidence interval:
      -Inf 0.9136743
sample estimates:
mean of x mean of y 
    5.032     4.661 




Crtl vs. Trt2 



	Welch Two Sample t-test

data:  PlantGrowth$weight[PlantGrowth$group == "ctrl"] and PlantGrowth$weight[PlantGrowth$group == "trt2"]
t = -2.134, df = 16.786, p-value = 0.02395
alternative hypothesis: true difference in means is less than 0
95 percent confidence interval:
        -Inf -0.09100612
sample estimates:
mean of x mean of y 
    5.032     5.526 


***For ctrl vs. Trt1: ***

The p-value = 0.8748. We cannot reject Ho. So the mean yield for the control group is not less than the mean yield from a plant given in  the treatments 1.

*** For ctrl vs. Trt2: ***

The p-value = 0.02395. We can reject Ho. So the mean yield for the control group is less than the mean yield from a plant given in  the treatments 2. 

*** As discussed, there is a rule of thumb for deciding whether to use a pooled estimate of the variance in an unpaired t-test. ***

*** e. Your task is to write a wrapper function that calls t.test after deciding whether it should be executed with var.equal=FALSE according to the rule of thumb. Use the following guidelines:*** 
* Your function should take four defined arguments: x and y with no defaults, to be treated in the same way as the same arguments in t.test; and var.equal and paired, with defaults that are the same as the defaults of t.test.
* An ellipsis (Section 9.2.5) should be included to represent any additional arguments to be passed to t.test.
* Upon execution, the function should determine whether paired=FALSE.
    - If paired is TRUE, then there is no need to proceed with the check of a pooled variance.
    - If paired is FALSE, then the function should determinethe value for var.equal automatically by using the rule of thumb.
* If the value of var.equal was set automatically, you can assume it will override any value of this argument initially supplied by the user.
* Then, call t.test appropriately.



In [26]:
wrapper <- function(x,y, var.equal = FALSE , paired = FALSE,...) {
    if(paired == TRUE) {
        t.test(x,y, paired = TRUE,var.equal = FALSE, ...)
    } 
    else {
        if(var.equal == FALSE) {
            t.test(x,y,paired = FALSE, var.equal = FALSE,...)
        } 
        else {
            t.test(x,y,paired = FALSE, var.equal = TRUE,...)
        }
    }
}

*** f. Try your new function on all three examples in the text of Section 18.2.2, ensuring you reach identical results. ***

In [23]:
snacks <- c(87.7,80.01,77.28,78.76,81.52,74.2,80.71,79.5,77.87,81.94,80.7,82.32,75.78,80.19,83.91,79.4,77.52,77.62,
            81.4,74.89,82.95, 73.59,77.92,77.18,79.83,81.23,79.28,78.44,79.01,80.47,76.23,
            78.89,77.14,69.94,78.54,79.7,82.45,77.29,75.52,77.21,75.99, 81.94,80.41,77.7)

snacks2 <- c(80.22,79.73,81.1,78.76,82.03,81.66,80.97,81.32,80.12,78.98, 79.21,81.48,79.86,81.06,77.96,80.73,
             80.34,80.01,81.82,79.3,79.08,79.47,78.98,80.87,82.24,77.22,80.03,79.2,80.95,79.17,81)

In [30]:
wrapper(x=snacks2, y=snacks, alternative = "greater", conf.level = 0.9)


	Welch Two Sample t-test

data:  x and y
t = 2.4455, df = 60.091, p-value = 0.008706
alternative hypothesis: true difference in means is greater than 0
90 percent confidence interval:
 0.5859714       Inf
sample estimates:
mean of x mean of y 
 80.15710  78.91068 


In [31]:
men <- c(102,87,101,96,107,101,91,85,108,67,85,82)
women <- c(73,81,111,109,143,95,92,120,93,89,119,79,90,126,62,92,77,106, 105,111)
wrapper(x=men, y=women, alternative = "two.sided", conf.level = 0.95, var.equal = TRUE)


	Two Sample t-test

data:  x and y
t = -0.93758, df = 30, p-value = 0.3559
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -19.016393   7.049727
sample estimates:
mean of x mean of y 
 92.66667  98.65000 


In [32]:
rate.before <- c(52,66,89,87,89,72,66,65,49,62,70,52,75,63,65,61)
rate.after <- c(51,66,71,73,70,68,60,51,40,57,65,53,64,56,60,59)
wrapper(x=rate.after,y=rate.before,alternative="less",conf.level=0.95, paired=TRUE)



	Paired t-test

data:  x and y
t = -4.8011, df = 15, p-value = 0.0001167
alternative hypothesis: true difference in means is less than 0
95 percent confidence interval:
      -Inf -4.721833
sample estimates:
mean of the differences 
                -7.4375 
